### PR TITLE
feat: 🎸 Replace old escrow contract

### PR DESCRIPTION
**Pull request by lovel8**
✅ Closes: #246
✅ Closes: #266 
## Related Issues

Due to the size limit of solidity deployment contract, the library needs to be changed to a separate contract deployment.

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

**Comment by waynewyang** [comment]
@lovel8  pls rebase
 @siriusyim  pls review this PR

File: src/v0.8/core/finance/base/Base.sol [hunk 1]
```diff
@@ -17,20 +17,51 @@
...
+import {MatchingType} from "src/v0.8/types/MatchingType.sol";
 
 import {ArraysPaymentInfoLIB} from "src/v0.8/shared/utils/array/ArrayLIB.sol";
 
-/// @title DatacapCollateralLIB
-/// @dev This library provides functions for managing DatacapCollateral-related operations.
-library DatacapCollateralLIB {
+/// @title Base
+/// @dev This Base provides functions for managing Escrow-related operations.
+contract Base is Initializable, UUPSUpgradeable, RolesModifiers {
```
Comment (siriusyim) [inline src/v0.8/core/finance/base/Base.sol:35]: Add the 'abstract' keyword

File: src/v0.8/module/dataset/Datasets.sol [hunk 2]
```diff
@@ -114,7 +120,19 @@ contract Datasets is
...
+        //     _datasetId,
+        //     0,
+        //     FinanceType.FIL,
+        //     FinanceType.Type.ChallengeCommission
+        // );
+
+        dataset._emitDatasetEvent(DatasetType.Event.Approved);
         emit DatasetsEvents.DatasetApproved(_datasetId);
+        // } else {
+        //     /// TODO: Add waitEscrow process(https://github.com/dataswap/core/issues/254)
```
Comment (siriusyim) [inline src/v0.8/module/dataset/Datasets.sol:134]: remove waitEscrow

File: src/v0.8/module/dataset/DatasetsProof.sol [hunk 3]
```diff
@@ -178,61 +178,39 @@ contract DatasetsProof is
         );
     }
 
-    ///@notice Submit proof for a dataset
```
Comment (siriusyim) [inline src/v0.8/module/dataset/DatasetsProof.sol:181]: This piece of code would be more appropriate at this location

**Review by siriusyim — CHANGES_REQUESTED** [review]

File: src/v0.8/core/finance/base/Base.sol (hunk 1 above)
Comment (lovel8) [reply src/v0.8/core/finance/base/Base.sol:35]: #277 It has been modified

File: src/v0.8/module/dataset/DatasetsProof.sol [hunk 4]
```diff
@@ -221,18 +221,39 @@ contract DatasetsProof is
...
+        return true;
+        // } else {
+        //     return false;
+        // }
+    }
+
     ///@notice Submit proof completed for a dataset
     function submitDatasetProofCompleted(
         uint64 _datasetId
-    )
```
Comment (siriusyim) [inline src/v0.8/module/dataset/DatasetsProof.sol:227]: Retain onlyDatasetState

**Review by siriusyim — CHANGES_REQUESTED** [review]

**Review by siriusyim — APPROVED** [review]

**Pull request merged**